src/linux/linux-host-session.vala

```diff
@@ -419,7 +408,47 @@ namespace Frida {
 			return result;
 		}
 
-		public async uint spawn (string package_name, string? activity_name) throws Error {
+		public async uint spawn (string program, HostSpawnOptions options) throws Error {
+
+			string package_name = program;
+
+			var aux_options = options.load_aux ();
+
+			string  method = "activity";
+			string? entry_point = null;
+			if (aux_options.contains ("activity")) {
+
+				string? activity_name = null;
+
+				if (!aux_options.lookup ("activity", "s", out activity_name))
+					throw new Error.INVALID_ARGUMENT ("The 'activity' option must be a string");
+
+				if (activity_name[0] == '.')
+					activity_name = package_name + activity_name;
+
+				entry_point = activity_name;
+			}
+			if (aux_options.contains ("receiver")) {
+
+				string? receiver_name = null;
+
+				if (!aux_options.lookup ("receiver", "s", out receiver_name))
+					throw new Error.INVALID_ARGUMENT ("The 'receiver' option must be a string");
+
+				if (receiver_name[0] == '.')
+					receiver_name = package_name + receiver_name;
+
+				entry_point = receiver_name;
+				method = "receiver";
+			}
+
+			string? action = null;
+			if (aux_options.contains ("action")) {
+

```

Comment:
Blank line should be dropped for consistency.

Reply:
ok